FirebaseDynamicLinks/Sources/FIRDynamicLink.m

```diff
@@ -134,6 +136,20 @@ + (FIRDLMatchType)matchTypeWithString:(NSString *)string {
   return [matchMap[string] integerValue] ?: FIRDLMatchTypeNone;
 }
 
++ (NSDictionary<NSString *, id> *)extractUtmParams:(NSDictionary<NSString *, id> *)parameters {
+  NSMutableDictionary<NSString *, id> *utmParamsDictionary = [[NSMutableDictionary alloc] init];
+
+  if (parameters) {
+    for (id key in parameters) {
+      if ([key hasPrefix:UTM_PARAM_PREFIX]) {
+        [utmParamsDictionary setObject:[parameters valueForKey:key] forKey:key];
+      }
+    }
+  }
+
+  return utmParamsDictionary;
```

Comment:
Return an immutable copy of the dictionary here, for safety.

Reply:
Made the change...please check whether this is what we are looking for.